### PR TITLE
make deployment of api on web server

### PR DESCRIPTION
**Pull request by coder2343**
This occurs with render at https://b4uagree.onrender.com. The api endpoints now will be https://b4uagree.onrender.com/sum, and https://b4uagree.onrender.com/sendpolicy

note we might need to ensure that render is configured properly in the parent repo after megrge is finished.

### CODE REVIEW

File: flask_controller/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Flask
 Gunicorn
 Torch
-bert-extractive-summarizer
```
Comment (Amani-Alqaisi) [inline flask_controller/requirements.txt]: assumed it was deleted by mistake then rewritten

File: flask_controller/app.py [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 
 app = Flask(__name__)
 
 @app.route('/')
 def hello_world():
   return 'Hello, World!'
 
+
+def get_summary(text,num_sentences):
+  """Function take text file and outputs given sumary of said text useing nlp model."""
```
Comment (Amani-Alqaisi) [inline flask_controller/app.py:13]: I'd add documentation next to the function since its the standard

File: flask_controller/app.py [hunk 3]
```diff
@@ -1,13 +1,21 @@
 from flask import Flask, request
 import json
 from nlp_code import get_summary
+from summarizer.sbert import SBertSummarizer
 
 app = Flask(__name__)
 
 @app.route('/')
 def hello_world():
   return 'Hello, World!'
```
Comment (Amani-Alqaisi) [inline flask_controller/app.py:9]: classic

File: flask_controller/nlp_code.py [hunk 4]
```diff
@@ -1,34 +1,8 @@
...
-    """Function take html file and extract given text."""
-    f = open("output/privacyPolicy.txt", "a", encoding="utf-8")
-    soup = BeautifulSoup(html, 'html.parser')
-    for header in soup.find_all({'h3','h2','h1'}):
-        next_node = header
-        while True:
-            next_node = next_node.nextSibling
-            if next_node is None:
-                break
-            if isinstance(next_node,NavigableString ):
```
Comment (Amani-Alqaisi) [inline flask_controller/nlp_code.py]: some odd spacing

File: flask_controller/app.py [hunk 5]
```diff
@@ -25,3 +23,6 @@ def get_privacy_policy():
 def send_summary():
   make_sum= get_summary(PrivacyPolicy.pop(),5)
```
Comment (Amani-Alqaisi) [inline flask_controller/app.py:31]: add a comment on what is number 5

File: flask_controller/app.py [hunk 6]
```diff
@@ -1,6 +1,27 @@
...
+
+PrivacyPolicy= []
+# function gets privacy policy
+@app.route('/sendpolicy', methods=['POST'])
+def get_privacy_policy():
+  request_data = request.get_json()
+  retrivedPolicy= request_data['privacyPolicy']
+  PrivacyPolicy.append(retrivedPolicy)
+  print(PrivacyPolicy)
+  return PrivacyPolicy
```
Comment (Amani-Alqaisi) [inline flask_controller/app.py:26]: why would it print and return

File: flask_controller/nlp_code.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+"""Module providing a function for sumerizing and bs4 is for webscrapeing."""
```
Comment (Amani-Alqaisi) [inline flask_controller/nlp_code.py]: Great header description

File: flask_controller/nlp_code.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+from summarizer.sbert import SBertSummarizer
+from bs4 import BeautifulSoup, NavigableString, Tag
+
+def html_to_text(html):
+    """Function take html file and extract given text."""
```
Comment (Amani-Alqaisi) [inline flask_controller/nlp_code.py]: side comment for in line is the standard, but good description

File: flask_controller/nlp_code.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        next_node = header
+        while True:
+            next_node = next_node.nextSibling
+            if next_node is None:
+                break
+            if isinstance(next_node,NavigableString ):
+                print (next_node.strip())
+            if isinstance(next_node, Tag):
+                if next_node.name == "h2"or next_node.name == "h1" or next_node.name == "h3" :
+                    #print(soup.find(string=nextNode.text.strip()))
```
Comment (Amani-Alqaisi) [inline flask_controller/nlp_code.py]: is this for testing or no longer works

File: flask_controller/nlp_code.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+def get_summary(text,num_sentences):
+    """Function take text file and outputs given sumary of said text useing nlp model."""
+    model = SBertSummarizer('paraphrase-MiniLM-L6-v2')
+    result = model(text, num_sentences=num_sentences)
+    return result
+
+def send_summary(result):
+    """Function take summary and saves text file"""
+    f = open("output/summary.txt", "a", encoding="utf-8")
```
Comment (Amani-Alqaisi) [inline flask_controller/nlp_code.py]: description of what the parameters are

File: flask_controller/flask_controller.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+from flask import Flask
+app = Flask(__name__)
```
Comment (Amani-Alqaisi) [inline flask_controller/flask_controller.py]: better naming or side comment description for app

**Review by Amani-Alqaisi — COMMENTED** [review]
The code review process has been completed for the selected unit of code. During the review, several aspects were addressed:

1. Functionality: We discussed the functionality provided to the users and ensured it meets the project requirements.

2. Readability: Comments were made regarding the clarity and ease of understanding of the code, aiming for improved readability.

3. Adherence to Coding Standards: We checked for adherence to the team's code style guide and made suggestions where necessary.

4. Suggestions for Improvement: There weren't any suggestions needed for improvement, the code and commits were straightforward, as expected since this is the flask folder.

5. Positive Aspects: positive aspects of the code were highlighted, acknowledging clear documentation.

Overall, the code review process has been valuable in enhancing the quality and maintainability of the codebase. We've documented our findings and recommendations for future reference.

File: flask_controller/app.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  model = SBertSummarizer('paraphrase-MiniLM-L6-v2')
+  result = model(text, num_sentences=num_sentences)
+  return result
+
+PrivacyPolicy= []
+# # function gets privacy policy
+@app.route('/sendpolicy', methods=['POST'])
+def get_privacy_policy():
+  request_data = request.get_json()
+  retrivedPolicy= request_data['privacyPolicy']
```
Comment (HotelCaliphornia) [inline flask_controller/app.py:23]: what is 'privacyPolicy' referring to in this line? ie. is that the name of a variable, is it matching this to text?

File: flask_controller/app.py [hunk 13]
```diff
@@ -0,0 +1,32 @@
+from flask import Flask, request
+import json
+from summarizer.sbert import SBertSummarizer
+
+app = Flask(__name__)
```
Comment (HotelCaliphornia) [inline flask_controller/app.py:5]: add comment documenting what this does

File: flask_controller/app.py [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+  return 'Hello, World!'
+
+
+def get_summary(text,num_sentences):
+  """Function take text file and outputs given sumary of said text useing nlp model."""
+  model = SBertSummarizer('paraphrase-MiniLM-L6-v2')
+  result = model(text, num_sentences=num_sentences)
+  return result
+
+PrivacyPolicy= []
```
Comment (HotelCaliphornia) [inline flask_controller/app.py:18]: should this be declared at the top of the document since its a global variable?

**Comment by HotelCaliphornia** [comment]
The code looks good, it accomplishes its aims quickly and concisely, only minor issues is some documentation stuff/general readability.

**Pull request merged**